source/fundamentals/auth.txt

```diff
@@ -0,0 +1,384 @@
+=========================
+Authentication Mechanisms
+=========================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+In this guide, we show you how to authenticate with MongoDB using each
+**authentication mechanism** available in the MongoDB Community Edition.
+Authentication mechanisms are processes by which the driver and server
+confirm identity and establish trust to ensure security.
+
+The mechanisms that you can use with the latest version of MongoDB Community
+Edition are as follows:
+
+* :ref:`Default <default-auth-mechanism>`
+* :ref:`SCRAM-SHA-256 <scram-sha-256-auth-mechanism>`
+* :ref:`SCRAM-SHA-1 <scram-sha-1-auth-mechanism>`
+* :ref:`MONGODB-CR <mongodb-cr-auth-mechanism>`
+* :ref:`MONGODB-AWS <mongodb-aws-auth-mechanism>`
+* :ref:`X.509 <x509-auth-mechanism>`
+
+If you need to authentication using ``Kerberos`` or ``LDAP`` see the
+:docs:`Enterprise Authentication Mechanisms guide <fundamentals/enterprise-auth>`.
+For more information on establishing a connection to your MongoDB cluster,
+read our :doc:`Connection Guide </fundamentals/connection>`.
+
+Specify an Authentication Mechanism
+-----------------------------------
+
+You can specify your authentication mechanism and credentials when connecting
+to MongoDB using either of the following:
+
+- a connection string
+- a ``MongoCredential`` factory method
+
+A **connection string** or **connection uri** provides instructions on how
+to connect and authenticate to your MongoDB cluster. To specify your
+authentication mechanism using a connection string, include your settings in
+your connection string and pass it to the ``MongoClients.create()`` method
+to instantiate your ``MongoClient``. For descriptions of how to use a
+connection string to specify your authentication method, navigate to the
+relevant mechanism section and select the :guilabel:`Connection String` tab.
+
+The ``MongoCredential`` class contains static factory methods that allow you
+to specify your authentication mechanism and credentials. You can use this
+helper class to keep your code organized. If you choose to use the
```

Comment:
`to keep your code organized`?  Not sure what you mean. Can you clarify?

Reply:
By "organized", I was referring to keeping specific pieces of logic in separate places that are identified by the code that wraps them. I labeled "organized" for lack of my ability to find a better word. 

I'm probably missing the advantages of using the helper class other than ensuring you're providing parameters that fit the method signature. If there any, can you explain them to me?

Reply:
You have it basically right.  The connection string is a grab bag of properties about everything from servers to credentials to timeouts to write concern.  It may not make sense to externalize all of them together in a single string (especially when the credentials are the one piece that have to remain secure).   As one example, some applications use an external credential provider which will set up temporary one-time credentials to use with MongoDB, that are provided to the application on startup via some sort of REST service.  So in that case you could have a connection string indicating the server names, and combine it with credentials that are retrieved at run time., e.g. 

```
   MongoClientSettings.builder()
       .applyConnectionString(...)
       .credential(<get credential at runtime>)
      .build()
```

Reply:
Ok, I'll give summarizing this a try.